src/smart-components/group/remove-group-modal.js

```diff
@@ -1,89 +1,112 @@
-import React, { useEffect } from 'react';
+import React, { useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
 import { withRouter } from 'react-router-dom';
 import { connect } from 'react-redux';
 import { bindActionCreators } from 'redux';
-import { Modal, Button, Grid, GridItem, Text, TextContent, TextVariants } from '@patternfly/react-core';
-import { fetchGroup, removeGroup } from '../../redux/actions/group-actions';
+import { Modal, Button, Text, TextContent, Checkbox } from '@patternfly/react-core';
+import { ExclamationTriangleIcon } from '@patternfly/react-icons';
+import { fetchGroup, removeGroup, removeGroups } from '../../redux/actions/group-actions';
 import { FormItemLoader } from '../../presentational-components/shared/loader-placeholders';
 
 const RemoveGroupModal = ({
   history: { goBack, push },
-  match: { params: { id }},
   removeGroup,
+  removeGroups,
   group,
   isLoading,
   fetchGroup,
+  groups,
+  isModalOpen,
   postMethod,
   closeUrl
 }) => {
   useEffect(() => {
-    fetchGroup(id);
+    if (groups.length === 1) {
+      fetchGroup(groups[0].uuid);
+    }
   }, []);
 
-  const onSubmit = () =>
-    postMethod ? removeGroup(id).then(() => postMethod([ id ])).then(push(closeUrl)) :
-      removeGroup(id).then(() => push(closeUrl));
+  const [ checked, setChecked ] = useState(false);
+
+  const multipleGroups = groups.length > 1;
+
+  const onSubmit = () => {
+    if (multipleGroups) {
+      let uuids = groups.map((group) => group.uuid);
+      removeGroups(uuids).then(() => postMethod()).then(push(closeUrl));
+    } else {
+      removeGroup([ groups[0].uuid ]).then(()=> postMethod()).then(push(closeUrl));
```

Comment:
I would have use `removeGroups` in here so we have on function to remove any amount of groups. Also, you probably forgot to pass UUIDs to postMethod. Also no need to use let if you are not reassigning it.
```JS
if (multipleGroups) {
  const uuids = groups.map((group) => group.uuid);
  removeGroups(uuids).then(() => postMethod(uuids)).then(push(closeUrl));
} else {
  removeGroups([ groups[0].uuid ]).then(([ groups[0].uuid ])=> postMethod()).then(push(closeUrl));
}
```

Reply:
Thanks! I removed that whole conditional statement!